### PR TITLE
Implement tests with Kitchen and Inspec, and CI with Travis

### PR DESCRIPTION
**Pull request by daks**
Centos7 disabled like for nginx-formula.
Opensuse42 disabled because of failed build. (will open an issue)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+source "https://rubygems.org"
+
+gem 'kitchen-docker', '>= 2.9'
+gem 'kitchen-salt', '>= 0.5.0'
```
Comment (myii) [inline Gemfile]: `0.5.0` => `0.6.0`

**Comment by myii** [comment]
As mentioned on Slack, got `opensuse` working with a sample entry for `Suse` in `map.jinja`.  See:

* https://github.com/myii/collectd-formula/commit/e8ca9fff94f40d3d4281ed629cd003bc1be499ce
* https://travis-ci.org/myii/collectd-formula/builds/519718387

@aboe76 Can you help us with the specific entries to be made in `map.jinja`?

**Comment by daks** [comment]
Fixes #136

**Comment by myii** [comment]
@aboe76 After some discussions in Slack, we've got `opensuse` working here as well.  Can you merge this if you're happy with it?  My only remaining concern is `map.jinja`, which I based on the `RedHat` entry.  The tests are passing but it's good to get another opinion.

@daks Thanks again for the excellent contribution -- review approved!
Shout out to @noelmcloughlin for the additional support.

**Comment by myii** [comment]
@aboe76 Any comments on this or can it be merged?  Specifically the concern about `map.jinja` as mentioned in the [comment above](https://github.com/saltstack-formulas/collectd-formula/pull/135#issuecomment-482888039)?

**Comment by aboe76** [comment]
@myii and @daks I don't see any issues, you are overwriting the defaults for Suse anyway so no issues.

**Comment by myii** [comment]
@aboe76 Thanks for the comments on the [commit in my branch](https://github.com/myii/collectd-formula/commit/e8ca9fff94f40d3d4281ed629cd003bc1be499ce).  I'll make those changes and then point @daks to them, to include here.

**Pull request merged**